app/Models/Entry.php

```diff
@@ -173,57 +168,51 @@ public function content(bool $withEnclosures = true, bool $allowDuplicateEnclosu
 			$length = $enclosure['length'] ?? 0;
 			$medium = $enclosure['medium'] ?? '';
 			$mime = $enclosure['type'] ?? '';
-			$thumbnails = $enclosure['thumbnails'] ?? [];
-			$etitle = $enclosure['title'] ?? '';
-
-			$content .= '<figure class="enclosure">';
-
-			foreach ($thumbnails as $thumbnail) {
-				$content .= '<p><img class="enclosure-thumbnail" src="' . $thumbnail . '" alt="" title="' . $etitle . '" /></p>';
-			}
+			$temp['thumbnails'] = $enclosure['thumbnails'] ?? [];
+			$temp['title'] = $enclosure['title'] ?? '';
 
 			if (self::enclosureIsImage($enclosure)) {
-				$content .= '<p class="enclosure-content"><img src="' . $elink . '" alt="" title="' . $etitle . '" /></p>';
+				$temp['type'] = 'image';
 			} elseif ($medium === 'audio' || strpos($mime, 'audio') === 0) {
-				$content .= '<p class="enclosure-content"><audio preload="none" src="' . $elink
-					. ($length == null ? '' : '" data-length="' . intval($length))
-					. ($mime == '' ? '' : '" data-type="' . htmlspecialchars($mime, ENT_COMPAT, 'UTF-8'))
-					. '" controls="controls" title="' . $etitle . '"></audio> <a download="" href="' . $elink . '">💾</a></p>';
+				$temp['type'] = 'audio';
+				$temp['data-length'] = $length == null ? '' : intval($length);
+				$temp['data-type'] = htmlspecialchars($mime, ENT_COMPAT, 'UTF-8');
 			} elseif ($medium === 'video' || strpos($mime, 'video') === 0) {
-				$content .= '<p class="enclosure-content"><video preload="none" src="' . $elink
-					. ($length == null ? '' : '" data-length="' . intval($length))
-					. ($mime == '' ? '' : '" data-type="' . htmlspecialchars($mime, ENT_COMPAT, 'UTF-8'))
-					. '" controls="controls" title="' . $etitle . '"></video> <a download="" href="' . $elink . '">💾</a></p>';
+				$temp['type'] = 'video';
+				$temp['data-length'] = $length == null ? '' : intval($length);
+				$temp['data-type'] = $mime == '' ? '' : htmlspecialchars($mime, ENT_COMPAT, 'UTF-8');
 			} else {	//e.g. application, text, unknown
-				$content .= '<p class="enclosure-content"><a download="" href="' . $elink
-					. ($mime == '' ? '' : '" data-type="' . htmlspecialchars($mime, ENT_COMPAT, 'UTF-8'))
-					. ($medium == '' ? '' : '" data-medium="' . htmlspecialchars($medium, ENT_COMPAT, 'UTF-8'))
-					. '" title="' . $etitle . '">💾</a></p>';
+				$temp['type'] = 'unknown';
+				$temp['data-medium'] = $medium == '' ? '' : htmlspecialchars($medium, ENT_COMPAT, 'UTF-8');
+				$temp['data-type'] = $mime == '' ? '' : htmlspecialchars($mime, ENT_COMPAT, 'UTF-8');
 			}
 
-			if ($credit != '') {
-				$content .= '<p class="enclosure-credits">© ' . $credit . '</p>';
-			}
-			if ($description != '') {
-				$content .= '<figcaption class="enclosure-description">' . $description . '</figcaption>';
-			}
-			$content .= "</figure>\n";
+			$this->content_enclosures[] = $temp;
 		}
 
 		return $content;
 	}
 
+	/**
+	 * @return array<int,string>|null
+	 */
+	public function content_enclosures(): array {
```

Comment:
@Alkarex @ColonelMoutarde I need your help. I cannot find a correct PHPstand comment for this Array. Could you please support here?

Reply:
go to https://framateam.org/freshrss

Reply:
Your return is always an array, not null